tests/encoding.rs

```diff
@@ -27,8 +27,8 @@ where
         buffer = vec![0u8; read_size];
 
         // Minimum size buffer assertion
-        d.read(buffer.as_mut_slice()).expect("Failed to read");
-        d_p.write(buffer.as_slice()).expect("Failed to write");
+        let _ = d.read(buffer.as_mut_slice()).expect("Failed to read");
```

Comment:
maybe allow this one per default? this one deteriorates the readability imo, and doesn't really make any sense for buffer contexts such as this one.